src/modules/twap/state/emulatedPartOrdersAtom.ts

```diff
@@ -0,0 +1,55 @@
+import { atom } from 'jotai'
+
+import { OrderClass, OrderStatus, SigningScheme } from '@cowprotocol/cow-sdk'
+
+import { OrderStatus as OrderStatusInApp } from 'legacy/state/orders/actions'
+
+import { OrderWithComposableCowInfo } from 'common/types'
+
+import { twapOrdersListAtom } from './twapOrdersListAtom'
+import { twapPartOrdersListAtom } from './twapPartOrdersAtom'
+
+import { TwapOrderItem, TwapOrderStatus } from '../types'
+
+export const emulatedPartOrdersAtom = atom<OrderWithComposableCowInfo[]>((get) => {
+  const twapOrders = get(twapOrdersListAtom)
+  const twapParticleOrders = get(twapPartOrdersListAtom)
+
+  return twapParticleOrders.map<OrderWithComposableCowInfo>((order) => {
+    const parent = twapOrders[order.twapOrderId]
```

Comment:
I guess no chance that you would get an undefined here?

Reply:
Only theoretically, if you manually change the store value in localStorage.
But in practice yes, there are no chances.